tests/pipeline/test_filter.py

```diff
@@ -379,3 +382,40 @@ def test_isfinite(self):
             initial_workspace={self.f: data},
         )
         check_arrays(results['isfinite'], isfinite(data))
+
+    def test_window_safe(self):
+        # all true data set of (days, securities)
+        data = full(self.default_shape, True, dtype=bool)
+
+        # rolling window length for TestFactor
+        k = 8
+
+        class InputFilter(Filter):
+            inputs = ()
+            window_length = 0
+
+        class TestFactor(CustomFactor):
+            dtype = float64_dtype
+            inputs = (InputFilter(), )
+            window_length = k
+
+            def compute(self, today, assets, out, filter_):
+                # sum for each column
+                out[:] = sum(filter_, axis=0)
```

Comment:
We should use `filter_.sum` rather than importing `sum` from numpy.  `sum` is a built-in, and it's generally a bad idea to alias builtins.  Another thing you could do would be to change the import to rebind sum to something else (e.g. `from numpy import sum as np_sum`) and then refer to it as `np_sum`.